packages/hooks/src/useSwitchToRealAccount.ts

```diff
@@ -0,0 +1,47 @@
+import { useStore } from '@deriv/stores';
+import React from 'react';
+import { routes } from '@deriv/shared';
+import { useHistory } from 'react-router-dom';
+
+const useSwitchToRealAccount = () => {
+    const { client, notifications, ui } = useStore();
+    const { account_list, has_active_real_account, is_virtual, is_pre_appstore, switchAccount, accounts, loginid } =
+        client;
+    const { showAccountSwitchToRealNotification } = notifications;
+    const { toggleReadyToDepositModal } = ui;
+    const history = useHistory();
+
+    const switchToReal = React.useCallback(() => {
+        if (is_pre_appstore && is_virtual && has_active_real_account) {
+            const new_account = account_list.find(account => account.loginid?.[0]);
+
+            if (new_account && new_account.loginid) {
+                const new_loginid = new_account.loginid;
+                const is_same_account = loginid === new_loginid;
+
+                if (!is_same_account) {
+                    switchAccount(new_loginid);
+                    showAccountSwitchToRealNotification(new_loginid, accounts[new_loginid].currency || 'USD');
+                }
+            }
+        } else if (is_pre_appstore && is_virtual && !has_active_real_account) {
+            history.push(routes.traders_hub);
```

Comment:
@thisyahlen-deriv Again, Not sure if it's a good idea to have navigation in business logic, Or at least we can break the `useSwitchToRealAccount` hook into smaller hooks with single responsibilities 🤔